src/plugins/discover/public/application/main/hooks/use_data_state.ts

```diff
@@ -10,17 +10,30 @@
 import { useState, useEffect } from 'react';
```

Comment:
Bug log:

1. `unsavedChanges` tag appears in many situations where it should not (only when Lens is part of the state).
    1. At the start, as we keep in state savedSearchId and abortComparator
    2. When we change the attribute of the dashboard that should not play role here, for example time range.
2.  When duplicating panel by value, the new title gets `(copy)` suffix (that’s expected). When duplicating panel by ref, the new title is the same. 
3. When you open the inline editor and cancel, the visualization disappears from the dashboard.
4. Open dashboard, move to edit mode, move to view mode, move to edit mode again. Try to open Lens context menu - it is impossible, it breaks with an error in the console.
5. Dirty state issue - Steps to reproduce:
	Create a new Lens visualization from the dashboard.
	Return to the dashboard after creating the visualization.
	Change the title of the visualization on the 
	Edit the same visualization again in the Lens 
	Click ‘Save and return’ to go back to the dashboard.

	Expected behavior: The title (and any embeddable prop, like a custom time range) should persist.
	Actual behavior: The title is wiped.